03-first-task.Rmd

```diff
@@ -417,4 +418,60 @@ task BwaMem {
     disks: "local-disk 100 SSD"
   }
 }
-  ```
\ No newline at end of file
+```
+
+## Putting the workflow together
+
+A workflow is needed to run the `BwaMem` task we just built. The workflow's input variables are defined by the workflow JSON metadata, and are then passed on as inputs in our `BwaMem` call. When the `BwaMem` call is complete, the workflow's output File variable is defined based on the task's output. Lastly, we have a parameter_meta component in our workflow that describes each workflow input variable as documentation.
+
+```         
+version 1.0
+
+workflow minidata_test_alignment {
+  input {
+    # Sample info
+    File sampleFastq
+    # Reference Genome information
+    File ref_fasta
+    File ref_fasta_index
+    File ref_dict
+    File ref_amb
+    File ref_ann
+    File ref_bwt
+    File ref_pac
+    File ref_sa
+  }
+
+  #  Map reads to reference
+  call BwaMem {
+    input:
+      input_fastq = sampleFastq,
+      ref_fasta = ref_fasta,
+      ref_fasta_index = ref_fasta_index,
+      ref_dict = ref_dict,
+      ref_amb = ref_amb,
+      ref_ann = ref_ann,
+      ref_bwt = ref_bwt,
+      ref_pac = ref_pac,
+      ref_sa = ref_sa
+  }
+   
+  # Outputs that will be retained when execution is complete
+  output {
+    File alignedBamSorted = BwaMem.analysisReadySorted
+  }
+
+  parameter_meta {
+    sampleFastq: "Filepath to sample .fastq"
```

Comment:
I'd change the parameter meta a bit. I think the "filepath" part is self-explanatory, and instead we should focus on information that people not familiar with the workflow may not expect, eg:

```
sampleFastq: "Sample .fastq (expects Illumina)"
ref_fasta: "Reference genome to align reads to"
ref_fasta_index: "Reference genome index"
```

And for the other reference genome stuff, I'd go with this general pattern to tell users not familiar with `_bwt` and friends where those files came from:
`ref_*: "Reference genome _* file (created by bwa index)"`